readthedocs/rtd_tests/tests/test_api.py

```diff
@@ -82,6 +87,27 @@ def test_make_build(self):
         self.assertEqual(build['output'], 'Test Output')
         self.assertEqual(build['state_display'], 'Cloning')
 
+    def test_restart_build(self):
+        build = get(
+            Build,
+            project=self.project,
+            version=self.version,
+        )
+        command = get(
+            BuildCommandResult,
+            build=build,
+        )
+        build.commands.add(command)
+
+        self.assertEqual(build.commands.count(), 1)
+
+        client = APIClient()
+        client.force_login(self.user)
+        r = client.post(reverse('build-restart', args=(build.pk,)))
+        self.assertEqual(r.status_code, 204)
+        self.assertEqual(build.commands.count(), 0)
```

Comment:
We should check all the other fields as well, no just the commands count (the ones that you are already cleaning like `cold_storage`, but also the ones that I'm suggesting)